src/modules/esl-media/core/esl-media.ts

```diff
@@ -205,6 +209,9 @@ export class ESLMedia extends ESLBaseElement {
         this.$$on(this._onConditionChange);
         this.deferredReinitialize();
         break;
+      case 'in-viewport':
+        this.inViewport ? this.play() : this.pause();
```

Comment:
should also relay on autoplay allowance
Probaly we need a special method to handle (in-viewport behavior + esl:show/esl:hide related changes)
(As an idea we can use the same attribute to make esl-media handle both scenarios hide due to viewport leave or hide due to esl:hide event)

I wasn't just assume that we can have troubles with playInVIewport feature. It is the same feature in general, observation of toggleable state is just less restrictive version of play-in-viewport that should handle only (visibility state change independently of viewport)